src/playlist-loader.js

```diff
@@ -52,59 +50,58 @@ const updateSegments = function(original, update, offset) {
-      if (playlist.segments &&
-          media.segments &&
-          playlist.segments.length === media.segments.length &&
-          playlist.mediaSequence === media.mediaSequence) {
-        continue;
-      }
+export const updateMaster = function(master, media) {
+  const result = mergeOptions(master, {});
+  const playlist = result.playlists.filter((p) => p.uri === media.uri)[0];
 
-      result.playlists[i] = mergeOptions(playlist, media);
-      result.playlists[media.uri] = result.playlists[i];
-
-      // if the update could overlap existing segment information,
-      // merge the two lists
-      if (playlist.segments) {
-        result.playlists[i].segments = updateSegments(
-          playlist.segments,
-          media.segments,
-          media.mediaSequence - playlist.mediaSequence
-        );
-      }
-      // resolve any missing segment and key URIs
-      j = 0;
-      if (result.playlists[i].segments) {
-        j = result.playlists[i].segments.length;
-      }
-      while (j--) {
-        segment = result.playlists[i].segments[j];
-        if (!segment.resolvedUri) {
-          segment.resolvedUri = resolveUrl(playlist.resolvedUri, segment.uri);
-        }
-        if (segment.key && !segment.key.resolvedUri) {
-          segment.key.resolvedUri = resolveUrl(playlist.resolvedUri, segment.key.uri);
-        }
-        if (segment.map && !segment.map.resolvedUri) {
-          segment.map.resolvedUri = resolveUrl(playlist.resolvedUri, segment.map.uri);
-        }
-      }
-      changed = true;
+  if (!playlist) {
+    return null;
+  }
+
+  // consider the playlist unchanged if the number of segments is equal and the media
+  // sequence number is unchanged
+  if (playlist.segments &&
+      media.segments &&
+      playlist.segments.length === media.segments.length &&
+      playlist.mediaSequence === media.mediaSequence) {
+    return null;
+  }
+
+  const mergedPlaylist = mergeOptions(playlist, media);
+
+  // if the update could overlap existing segment information, merge the two segment lists
+  if (playlist.segments) {
+    mergedPlaylist.segments = updateSegments(
+      playlist.segments,
+      media.segments,
+      media.mediaSequence - playlist.mediaSequence
+    );
+  }
+
+  // resolve any segment URIs to prevent us from having to do it later
+  mergedPlaylist.segments.forEach((segment) => {
+    if (!segment.resolvedUri) {
+      segment.resolvedUri = resolveUrl(mergedPlaylist.resolvedUri, segment.uri);
+    }
+    if (segment.key && !segment.key.resolvedUri) {
+      segment.key.resolvedUri = resolveUrl(mergedPlaylist.resolvedUri, segment.key.uri);
+    }
+    if (segment.map && !segment.map.resolvedUri) {
+      segment.map.resolvedUri = resolveUrl(mergedPlaylist.resolvedUri, segment.map.uri);
+    }
+  });
+
+  // TODO Right now in the playlists array there are two references to each playlist, one
+  // that is referenced by index, and one by URI. The index reference may no longer be
```

Comment:
I'm pretty sure everywhere within contrib-hls uses playlists as an array and would require that the index based referenced be updated in order to get the new segments

Reply:
It might be worth considering, but can take out the TODO for now so it doesn't produce excess noise in the linter.